PetGame/Services/PetService.cs

```diff
@@ -200,5 +211,196 @@ public bool DeletePet(ulong petId, ulong ownerId)
+            //happiness starts at 100%
+            double happinessPercentage = 1.0;
+
+            DateTime ToCheck = DateTime.Now.Subtract(TimeSpan.FromDays(NumDaysToCheck));
+            int HoursToCheck = NumDaysToCheck * 24;
+
+            List<Activity> PastActivities = (List<Activity>)activityService.GetActivities(PetId, NumActivities, ToCheck, null);
+
+            //calculate hunger
+            //if there are no activities, dcrease hunger by number of hours
+            if (PastActivities.Count == 0)
+            {
+                hungerPercentage -= (HungerDecreasePerHour * HoursToCheck);
+            }
+            else
+            {
+                //check the type of activity and subtract percentage accordingly
+                foreach (Activity Activity in PastActivities)
+                {
+                    //if a Feeding activity has occured, increase hunger
+                    if (Activity.Type.Equals(ActivityType.Feeding))
+                    {
+                        if (hungerPercentage < 1.0 &&
+                            ((hungerPercentage += HungerIncrease) <= 1.0))
+                        {
+                            hungerPercentage += HungerIncrease;
+                        }
+                        else
+                        {
+                            hungerPercentage = 1.0;
+                        }
+                    }
+                    //if a Training event has occured, subtract
+                    else if (Activity.Type.Equals(ActivityType.Training))
+                    {
+                        if (hungerPercentage > 0 &&
+                            ((hungerPercentage -= TrainingDecrease) > 0))
+                        {
+                            hungerPercentage -= TrainingDecrease;
+                        }
+                        else
+                        {
+                            hungerPercentage = 0;
+                        }
+                    }
+                    //if a Race event has occurred
+                    else if (Activity.Type.Equals(ActivityType.Race))
+                    {
+                        if (hungerPercentage > 0 &&
+                            ((hungerPercentage -= RaceDecrease) > 0))
+                        {
+                            hungerPercentage -= RaceDecrease;
+                        }
+                        else
+                        {
+                            hungerPercentage = 0;
+                        }
+                    }
+                    //else, decrease hunger
+                    else
+                    {
+                        if (hungerPercentage > 0 &&
+                            ((hungerPercentage -= HungerDecreasePerHour) > 0))
+                        {
+                            hungerPercentage -= HungerDecreasePerHour;
+                        }
+                        else
+                        {
+                            hungerPercentage = 0;
+                        }
```

Comment:
You repeat yourself a lot here, and arguably each of these actions could be broken into their own method.